bigtable_rs/src/bigtable.rs

```diff
@@ -296,6 +296,61 @@ impl BigTableConnection {
         }
     }
 
+    pub async fn new_with_path(
+        project_id: &str,
+        instance_name: &str,
+        is_read_only: bool,
+        channel_size: usize,
+        timeout: Option<Duration>,
+        credential_path: &str,
+    ) -> Result<Self> {
+        let authentication_manager = AuthenticationManager::from(CustomServiceAccount::from_file(credential_path)?);
+        
```

Comment:
@kiv1 Thanks. Perhaps I didn't make myself clear enough. I was thinking about making a new method and call it like `new_with_auth_manager(..., authentication_manager: AuthenticationManager)`, so the user can create one and passing in whenever needed. Then the changes here will be very simple as we just need to reuse the code in the `new()` methods.

Let me write some code here and see if you think making sense:

```rust
// change the old new() method to something like this:
pub async fn new_with_auth_manager(
        project_id: &str,
        instance_name: &str,
        is_read_only: bool,
        channel_size: usize,
        timeout: Option<Duration>,
        authentication_manager: AuthenticationManager,
    ) -> Result<Self> {
  // here the code will be mostly the same with the old new() method, but just
  // removing the line of
  // let authentication_manager = AuthenticationManager::new().await?;
  ...
}

// then create a new new() method and internally call new_with_auth_manager()
pub async fn new(
        project_id: &str,
        instance_name: &str,
        is_read_only: bool,
        channel_size: usize,
        timeout: Option<Duration>,
    ) -> Result<Self> {
  let authentication_manager = AuthenticationManager::new().await?;
  
  new_with_auth_manager(
    project_id, instance_name, is_read_only, channel_size, timeout, authentication_manager
  )
}
```

Then this change will be very simple and no need to duplicate code there. 

Let me know how you think :)

Reply:
So sorry about that! Understood and updated accordingly.

Reply:
Nothing to sorry for, thanks a lot :)